src/test/bridges/erc4626/ERC4626.t.sol

```diff
@@ -112,13 +112,30 @@ contract ERC4626Test is BridgeTestBase {
             assertEq(outputValueB, 0, "Non-zero outputValueB");
             assertFalse(isAsync, "Bridge is not synchronous");
 
+            uint256 processorShareBalanceBeforeRedeem = IERC20(shares[i].erc20Address).balanceOf(
+                address(ROLLUP_PROCESSOR)
+            );
+
             // Immediately redeem the shares
-            uint256 redeemAmount = outputValueA;
+            uint256 redeemAmount = bound(_shareAmount, 1, outputValueA);
+
+            // expectedAssetAmountReturned = sharesRedeemed/sharesMinted * inputAssetAmount
+            uint256 expectedAssetAmountReturned = (redeemAmount * assetAmount) / outputValueA;
 
             bridgeCallData = encodeBridgeCallData(id, shares[i], emptyAsset, assets[i], emptyAsset, 1);
             (outputValueA, outputValueB, isAsync) = sendDefiRollup(bridgeCallData, redeemAmount);
 
-            assertApproxEqAbs(outputValueA, assetAmount, 2, "Received amount of asset differs from the expected one");
+            assertEq(
+                IERC20(shares[i].erc20Address).balanceOf(address(ROLLUP_PROCESSOR)) + redeemAmount,
+                processorShareBalanceBeforeRedeem,
+                "Incorrect RollupProcessor share balance after redeem"
+            );
+            assertApproxEqAbs(
```

Comment:
Should be possible to have exact.

Reply:
Addressed in f939b9b47721ee94ce35084ac073581650e5c3bf